frontend/src/components/viewers/MetricsVisualizations.tsx

```diff
@@ -593,7 +579,24 @@ export function ConfidenceMetricsSection({
   filter,
 }: ConfidenceMetricsSectionProps) {
   const maxSelectedRocCurves: number = 10;
+  const [allLinkedArtifacts, setAllLinkedArtifacts] = useState<LinkedArtifact[]>(linkedArtifacts);
   const [linkedArtifactsPage, setLinkedArtifactsPage] = useState<LinkedArtifact[]>(linkedArtifacts);
```

Comment:
What is difference between `allLinkedArtifacts ` and `linkedArtifactsPage `? Do we need two states for them?

Reply:
`allLinkedArtifacts` stores the entire list of `linkedArtifacts` passed into this component `ConfidenceMetricsSection`, and I use this to determine whether the `linkedArtifacts` list has changed (if it has, then I `return null` and wait for the state variable refresh).

The `linkedArtifactsPage` stores just the list of linked artifacts which are present on the present selected page of the filter table (by default, this has a maximum of `10`). I need this variable separately in order to keep track of which artifacts are currently displayed, but since this page of information alone cannot signal whether the `linkedArtifacts` prop has changed, I added the `allLinkedArtifacts` prop.

Reply:
I might be wrong but you are observing `[linkedArtifacts]` in `useEffect()` below. Can I assume that each time `useEffect`  is triggered, there is a change in `linkedArtifacts`?

Reply:
Yes, I think that is fair. I can remove the `!isEqual(linkedArtifacts, allLinkedArtifacts)` from inside the `useEffect` (I added for thoroughness but think it is unnecessary, done here: ) but I still need to `return null` when the artifacts differ. I could potentially do this by searching through the `filter` variables to see if any of them hold a reference to an artifact from a de-selected run, but this method of adding a state variable seemed a bit simpler.

Should I try to avoid adding that state variable, maybe adding some further logic to identify when the `linkedArtifacts` changes instead? I don't think I can `return null` from the `useEffect` so I wanted to find a way to identify that error outside of that hook.

Reply:
I think the main idea to introduce `allLinkedArtifacts` here is to avoid error being surfaced during transitioning state. It sounds good with your current solution which is more straightforward.